channel/handlers_server.go

```diff
@@ -1606,3 +1612,69 @@ func (server *Server) playerPetLoot(conn mnet.Client, reader mpacket.Reader) {
 
 	plr.inst.dropPool.playerAttemptPickup(drop, plr, 5)
 }
+
+func (server *Server) handleCharacterDeleted(conn mnet.Server, reader mpacket.Reader) {
+	charID := reader.ReadInt32()
+
+	// Remove from any party
+	for _, party := range server.parties {
+		for i, plr := range party.players {
+			if plr != nil && plr.ID == charID {
+				// Remove from party
```

Comment:
There already is a party remove function: https://github.com/Hucaru/Valhalla/blob/master/channel/party.go#L50. However, it will need to be refactored to remove the Deserialisation (I named the function incorrectly it looks like) into it's parent call.